api/src/database/database.py

```diff
@@ -50,34 +51,52 @@ def is_connected(self):
         Checks the connection status
         :return: True if the database is accessible False otherwise
         """
-        return self.engine is not None and self.session is not None
+        return self.engine is not None and global_session is not None
 
     def start_session(self):
         """
-        Starts a session
-        :return: True if the session was started, False otherwise
+        :return: Database singleton session
         """
+        global global_session
+        try:
+            if global_session is not None:
+                self.logger.info("Database session reused.")
+                return global_session
+            if global_session is None or not global_session.is_active:
+                global_session = self.start_new_db_session()
+                self.logger.info("Global Singleton Database session started.")
+                return global_session
+        except Exception as error:
+            raise Exception(f"Error creating database session: {error}")
+
+    def start_new_db_session(self):
+        global global_session
         try:
-            if self.engine is None:
-                self.connection_attempts += 1
-                self.logger.debug(f"Database connection attempt #{self.connection_attempts}.")
+            lock.acquire()
+            if global_session is not None and global_session.is_active:
+                self.logger.info("Database session reused.")
+                return global_session
+            if self.SQLALCHEMY_DATABASE_URL is None:
+                raise Exception("Database URL is not set")
+            else:
+                self.logger.info("Starting new global database session.")
                 self.engine = create_engine(self.SQLALCHEMY_DATABASE_URL, echo=True)
-                self.logger.debug("Database connected.")
-            if self.session is not None and self.session.is_active:
-                self.session.close()
-            self.session = Session(self.engine, autoflush=False)
-        except Exception as e:
-            self.logger.error(f"Database new session creation failed with exception: \n {e}")
-        return self.is_connected()
+                global_session = sessionmaker(bind=self.engine)()
+                self.session = global_session
```

Comment:
I think the _self.session_ is only used in this assignment as it's replaced by accessing the _ global_session_ directly. If this is true, let's remove this assignment.